### ISSUE TITLE
SFPx extension completely breaks if you delete from a site, update, and try to re-add

### ISSUE THREAD

**Issue by corriespondent** [SharePoint/sp-dev-docs#4362]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

I would expect that you could delete an SPFx extension (tenant deployed, but not tenant scoped) from a site collection and then re-add it without any issues.

#### Observed Behavior

When I delete an SPFx extension from a site collection (via Site Contents), delete from the recycle bin, update the extension, and then try to add it back, the extension does not work any longer.

#### Steps to Reproduce

1. I had a header extension deployed in my tenant's app catalog, and had added it (via Add an App) to a specific site collection. The extension is working on the site at this point, as in, I can see the custom header in the header placeholder area, and properties are passed in.
2. Removed app from Site Contents (had to change to classic SharePoint to do this).
Header no longer shows.
3. Cleared cookies to return to Modern SharePoint to the Site Contents page of my site collection, so that I could go delete it from the recycle bin as well.
4. I deployed a new version of the solution package.
5. Added the app back again through the UI. This has seemed to irrevocably break it.  Although the app shows in the site contents, the extension doesn't work, and Get-PnPCustomAction returns nothing when I test it on the site collection. I have not found any way to get the extension working again.

**Comment by patmill** [comment]
Is there any chance that the site has a records-management hold on it?

**Comment by corriespondent** [comment]
@patmill the site did not have a records-management hold on it, but I think I figured out the issue! There were two steps I was missing, which were not intuitive:

- I did not realize there was a secondary recycle bin where I needed to also delete from as well
- I also needed to click on the new app's ... menu, go to Monitor, go to the App Profile, and see that there was a new version available and click the "Get it" button.

I think this issue can be closed, thanks much.
